baseclasses/pyAero_problem.py

```diff
@@ -449,13 +473,67 @@ def _setStates(self, inputDict):
                 "to correctly specify the aerodynamic state"
             )
 
-    def setBCVar(self, varName, value, familyName):
+    def setBCVar(self, varName, value, groupName):
         """
         set the value of a BC variable on a specific variable
         """
+        if not groupName in self.bc_data.keys():
+            self.bc_data[groupName] = {}
+
+        self.bc_data[groupName][varName] = value
+
+    def getBCData(self):
+        return self.bc_data
+
+    # def setBCArray(self, groupName, varName, dataArray, patch=None):
```

Comment:
Is this needed? I guess this is related to the "nested" approach below, I am wondering for which cases having an array as an input would make sense